src/FileServiceManager/FileService/DAL/Storages/FileSystemStorage/FileSystemStorage.cs

```diff
@@ -0,0 +1,132 @@
+﻿using FileService.Models;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.IO;
+using FileService.Helpers;
+using FileService.DAL.Storages.Options;
+using Microsoft.Extensions.Options;
+
+namespace FileService.DAL
+{
+    public class FileSystemStorage : IStorage
+    {
+        private readonly string _basePath;
+        private readonly int _fileSizeLimit;
+        private readonly FileExtension[] _allowedExtensions;
+        private readonly IFileSystemConfigurator _fileSystemConfigurator;
+
+        public FileSystemStorage(IOptions<StorageOption> options, IFileSystemConfigurator fileSystemConfigurator)
+        {
+            _basePath = Path.Combine((string.IsNullOrEmpty(options.Value.Path) ? Directory.GetCurrentDirectory() : options.Value.Path) + "\\Files\\");
```

Comment:
why this Files part of the Path is not in the Configuration itself ?

Reply:
To make sure that we always have Files catalog. I can remove it though, it depends on our convention